signup.js

```diff
@@ -0,0 +1,92 @@
+//이메일 형식
+const myEmail = "test@codeit.com";
+
+function validateEmail(email) {
+  const regex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
+  return regex.test(email);
+}
+
+// 이메일 유효성 검사
+const emailInput = document.querySelector(".sign-input");
+const errorEmail = document.querySelector("#error-message-email");
+
+emailInput.addEventListener("focusout", function () {
+  const email = emailInput.value;
+
+  if (email === "") {
+    errorEmail.innerHTML = "이메일을 입력하세요.";
+    return;
+  }
+  if (!validateEmail(email)) {
+    errorEmail.innerHTML = "올바른 이메일 주소가 아닙니다.";
+    return;
+  }
+  if (validateEmail(email)) {
+    errorEmail.innerHTML = "";
+  }
+  // 이미 존재하는 이메일.
+  if (email === myEmail) {
+    errorEmail.innerHTML = "이미 사용 중인 이메일입니다.";
+  }
+});
```

Comment:
ealry return을 사용하여 로직을 잘 짜주셨군요 너무 좋습니다!

여기서 개선을 조금 더 해보자면 바깥의 emailInput을 받는게 아니라 event 핸들러 사용시에 자동으로 부여되는 `event` 객체를 활용하면 어떨까요?? 👀👀👀

```suggestion
emailInput.addEventListener("focusout", function (event) {
  const email = event.target.value;

  if (email === "") {
    errorEmail.innerHTML = "이메일을 입력하세요.";
    return;
  }
  if (!validateEmail(email)) {
    errorEmail.innerHTML = "올바른 이메일 주소가 아닙니다.";
    return;
  }
  if (validateEmail(email)) {
    errorEmail.innerHTML = "";
  }
  // 이미 존재하는 이메일.
  if (email === myEmail) {
    errorEmail.innerHTML = "이미 사용 중인 이메일입니다.";
  }
});
```